File: var/web/TestCgiSite/action_page.py
import os

# print("Content-type: text/html\r\n\r")
# Print HTML content
print("<!DOCTYPE html>")
print("<html>")
print("<head>")
print("<title>Environment Variables</title>")
print("</head>")
print("<body>")
print("<h1>Environment Variables:</h1>")
print("<ul>")
for key, value in os.environ.items():
    print(f"<li><strong>{key}:</strong> {value}</li>")
print("</ul>")
print("</body>")
print("</html>")

File: var/web/TestCgiSite/action_page.php
<!DOCTYPE html>
<html>
<head>
    <title>QUERY_STRING Output</title>
</head>
<body>

<h1>QUERY_STRING Output</h1>

<div>
    <?php
    // Check if QUERY_STRING is set
    // if (isset($_SERVER['QUERY_STRING'])) {
    //     // Get and display the value of QUERY_STRING
    //     $queryString = $_SERVER['QUERY_STRING'];
    //     echo "<p>QUERY_STRING: $queryString</p>";
    // } else {
    //     echo "<p>QUERY_STRING is not set.</p>";
    // }
    foreach ($_SERVER as $key => $value) {
        echo $key . " => " . $value . "<br>";
    }
    ?>
</div>
</body>
</html>